test/e2e/e2e.go

```diff
@@ -188,14 +189,48 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
 
 })
 
+var lock sync.Mutex
+var cleanupActions = map[*int]func(){}
+
+func AddCleanupAction(fn func()) *int {
+	p := new(int)
```

Comment:
I want a handle that is guaranteed unique - pointers seems to be cheap and easy.  I'll put it behind a type.